python/graphstorm/gconstruct/transform.py

```diff
@@ -190,6 +190,13 @@ def process_features(data, ops):
         res = op(data[op.col_name])
         assert isinstance(res, dict)
         for key, val in res.items():
+            # Check if has 1D features. If yes, convert to 2D features
+            if len(val.shape) == 1:
+                if isinstance(val, HDF5Array):
+                    val = val.to_numpy().reshape(-1, 1)
```

Comment:
it's better to print a message here saying that "We need to load the data from HDF5 to memory to convert its shape to XX".

The reason that we have HDF5Array is to avoid loading its data to memory during the graph construction to reduce memory consumption.